packages/asset-swapper/src/utils/quote_requestor.ts

```diff
@@ -0,0 +1,99 @@
+import { assetDataUtils, SignedOrder } from '@0x/order-utils';
+import { ERC20AssetData } from '@0x/types';
+import { BigNumber, logUtils } from '@0x/utils';
+import Axios, { AxiosResponse } from 'axios';
+
+import { MarketOperation } from '../types';
+
+/**
+ * Request quotes from RFQ-T providers
+ */
+export class QuoteRequestor {
+    private readonly _rfqtMakerEndpoints: string[];
+
+    constructor(rfqtMakerEndpoints: string[]) {
+        this._rfqtMakerEndpoints = rfqtMakerEndpoints;
+    }
+
+    public async requestRfqtFirmQuotesAsync(
+        makerAssetData: string,
+        takerAssetData: string,
+        assetFillAmount: BigNumber,
+        marketOperation: MarketOperation,
+        intentOnFilling: boolean,
+        takerApiKey: string,
+        takerAddress: string,
+    ): Promise<SignedOrder[]> {
+        const makerEndpointMaxResponseTimeMs = 1000;
+
+        const getTokenAddressOrThrow = (assetData: string): string => {
+            const decodedAssetData = assetDataUtils.decodeAssetDataOrThrow(assetData);
+            if (decodedAssetData.hasOwnProperty('tokenAddress')) {
+                // type cast necessary here as decodeAssetDataOrThrow returns
+                // an AssetData object, which doesn't necessarily contain a
+                // token address.  (it could possibly be a StaticCallAssetData,
+                // which lacks an address.)  so we'll just assume it's a token
+                // here.  should be safe, with the enclosing guard condition
+                // and subsequent error.
+                // tslint:disable-next-line:no-unnecessary-type-assertion
+                return (decodedAssetData as ERC20AssetData).tokenAddress;
+            } else {
+                throw new Error(
+                    `Decoded asset data (${JSON.stringify(decodedAssetData)}) does not contain a token address`,
+                );
+            }
+        };
+
+        const buyToken = getTokenAddressOrThrow(makerAssetData);
+        const sellToken = getTokenAddressOrThrow(takerAssetData);
+
+        // create an array of promises for quote responses, using "undefined"
+        // as a placeholder for failed requests.
+
+        const responsePromises: Array<Promise<AxiosResponse<SignedOrder> | undefined>> = [];
+        for (const rfqtMakerEndpoint of this._rfqtMakerEndpoints) {
+            responsePromises.push(
+                Axios.get(`${rfqtMakerEndpoint}/quote`, {
+                    headers: { '0x-api-key': takerApiKey },
+                    params: {
+                        sellToken,
+                        buyToken,
+                        buyAmount: marketOperation === MarketOperation.Buy ? assetFillAmount.toString() : undefined,
```

Comment:
We'll be telling external makers that they should only focus on serving quotes with buyAmount.  Can optimize latency here by only hitting quote providers if a buyAmount was provided; why waste time on a sellAmount request if we know quote providers won't give a quote anyways.

Reply:
Why do makers prefer buys over sells?

Reply:
Also interested to hear the preference for `buyAmount`. I would consider them both equally required, or if only one can be provided, then `sellAmount`

Reply:
I misspoke in the initial comment here.  We're recommending serving quotes with sellAmount, not buyAmount.

We're just focusing on sells because buys are less than 20% of transactions, because many liquidity providers don't support buys, so we've been driving integrators towards sells.  We can expand later and support buys in the future.  Just reasoning about sells will be easier for market makers.